clusterloader2/pkg/measurement/common/simple/service_creation_latency.go

```diff
@@ -0,0 +1,422 @@
+	pingChecks                           = 10
+)
+
+func init() {
+	measurement.Register(serviceCreationLatencyName, createServiceCreationLatencyMeasurement)
+}
+
+func createServiceCreationLatencyMeasurement() measurement.Measurement {
+	return &serviceCreationLatencyMeasurement{
+		queue:         workerqueue.NewWorkerQueue(serviceCreationLatencyWorkers),
+		creationTimes: make(map[string]*serviceCreationTimes),
+		pingCheckers:  make(map[string]*pingChecker),
+	}
+}
+
+type serviceCreationLatencyMeasurement struct {
+	namespace        string
+	labelSelector    string
+	fieldSelector    string
+	waitTimeout      time.Duration
+	stopCh           chan struct{}
+	isRunning        bool
+	queue            workerqueue.Interface
+	informer         cache.SharedInformer
+	clusterFramework *framework.Framework
+
+	lock          sync.Mutex
+	creationTimes map[string]*serviceCreationTimes
+	pingCheckers  map[string]*pingChecker
+}
+
+// Execute waits until services are ready.
+// Services can be specified by field and/or label selectors.
+// If namespace is not passed by parameter, all-namespace scope is assumed.
+// "start" action starts observation of the services.
+// "waitTillReady" waits until all services are available.
+// "gather" returns service creation latency summary.
+func (s *serviceCreationLatencyMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+	s.clusterFramework = config.ClusterFramework
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return nil, err
+	}
+
+	switch action {
+	case "start":
+		s.namespace, err = util.GetStringOrDefault(config.Params, "namespace", metav1.NamespaceAll)
+		if err != nil {
+			return nil, err
+		}
+		s.labelSelector, err = util.GetStringOrDefault(config.Params, "labelSelector", "")
+		if err != nil {
+			return nil, err
+		}
+		s.fieldSelector, err = util.GetStringOrDefault(config.Params, "fieldSelector", "")
+		if err != nil {
+			return nil, err
+		}
+		s.waitTimeout, err = util.GetDurationOrDefault(config.Params, "waitTimeout", defaultServiceCreationLatencyTimeout)
+		if err != nil {
+			return nil, err
+		}
+		return nil, s.start()
+	case "waitTillReady":
+		return nil, s.waitTillReady()
+	case "gather":
+		return s.gather()
+	default:
+		return nil, fmt.Errorf("unknown action %v", action)
+	}
+}
+
+// Dispose cleans up after the measurement.
+func (s *serviceCreationLatencyMeasurement) Dispose() {
+	if s.isRunning {
+		s.isRunning = false
+		close(s.stopCh)
+	}
+	s.queue.Stop()
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	for _, pc := range s.pingCheckers {
+		pc.stop()
+	}
+	s.pingCheckers = make(map[string]*pingChecker)
+}
+
+// String returns a string representation of the metric.
+func (*serviceCreationLatencyMeasurement) String() string {
+	return serviceCreationLatencyName
+}
+
+type serviceCreationTimes struct {
+	creation  time.Time
+	assigned  time.Time
+	available time.Time
```

Comment:
reachable?

[In general naming is strange here - you have:
- creation - noun
- assigned/available - not nouns
Should be unified.]

Reply:
Done.